### ISSUE TITLE
nf-core lint no longer respects skipping pipeline name checks

### ISSUE THREAD

**Issue by awgymer** [nf-core/tools#2027]
### Description of the bug

Due to the addition of a check on pipeline name added to `PipelineCreate.create_param_dict` attempting to lint a pipeline but skip linting the name no longer works as reported by @matthdsm on Slack. 

Potential solutions:
* remove check from method
* leverage `.nf-core.yml` to selectively ignore this check.

If we go option 2 then I would propose making tools always attempt to load the `.nf-core.yml` (we are already possibly expanding where it is needed due to path changes) and store the config in some globally available object that can be easily referred to across the tools code. 

### Command used and terminal output

```console
nf-core lint

    nf-core/tools version 2.7.dev0 - https://nf-co.re


INFO     Testing pipeline: .                                                                                                                   
Running lint checks ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0 of 18 » files_existERROR    Invalid workflow name: must be lowercase without punctuation.
```

### System information

nf-core/tools 2.7.dev0

**Comment by mirpedrol** [comment]
Fixed in #2035

**Comment by matthdsm** [comment]
Thanks @mirpedrol
